### PR TITLE
Tests + backport fix for teleport_assets (ReceiveTeleportedAsset)

### PR DESCRIPTION
**Pull request by bkontur**

### CODE REVIEW

File: parachains/runtimes/assets/statemine/src/weights/xcm/mod.rs [hunk 1]
```diff
@@ -35,7 +35,18 @@ impl WeighMultiAssets for MultiAssetFilter {
...
+				All => weight.saturating_mul(MAX_ASSETS as u64),
+				AllOf { fun, .. } => match fun {
+					WildFungibility::Fungible => weight,
+					// Magic number 2 has to do with the fact that we could have up to 2 times
+					// MaxAssetsIntoHolding in the worst-case scenario.
+					WildFungibility::NonFungible =>
+						weight.saturating_mul((MaxAssetsIntoHolding::get() * 2) as u64),
+				},
+				AllCounted(count) => weight.saturating_mul(*count as u64),
+				AllOfCounted { count, .. } => weight.saturating_mul(*count as u64),
```
Comment (KiChjang) [inline parachains/runtimes/assets/statemine/src/weights/xcm/mod.rs:48]:
```suggestion
				AllCounted(count) => weight.saturating_mul(MAX_ASSETS.min(*count as u64)),
				AllOfCounted { count, .. } => weight.saturating_mul(MAX_ASSETS.min(*count as u64)),
```

We need to ensure that people don't try to use an outrageously large number that may cause a crazily high weight. Make sure to also change the type of `MAX_ASSETS` to `u64`.

File: parachains/runtimes/assets/statemint/src/weights/xcm/mod.rs [hunk 2]
```diff
@@ -35,7 +35,18 @@ impl WeighMultiAssets for MultiAssetFilter {
...
+				All => weight.saturating_mul(MAX_ASSETS as u64),
+				AllOf { fun, .. } => match fun {
+					WildFungibility::Fungible => weight,
+					// Magic number 2 has to do with the fact that we could have up to 2 times
+					// MaxAssetsIntoHolding in the worst-case scenario.
+					WildFungibility::NonFungible =>
+						weight.saturating_mul((MaxAssetsIntoHolding::get() * 2) as u64),
+				},
+				AllCounted(count) => weight.saturating_mul(*count as u64),
+				AllOfCounted { count, .. } => weight.saturating_mul(*count as u64),
```
Comment (KiChjang) [inline parachains/runtimes/assets/statemint/src/weights/xcm/mod.rs:48]:
```suggestion
				AllCounted(count) => weight.saturating_mul(MAX_ASSETS.min(*count as u64)),
				AllOfCounted { count, .. } => weight.saturating_mul(MAX_ASSETS.min(*count as u64)),
```

File: parachains/runtimes/assets/westmint/src/weights/xcm/mod.rs [hunk 3]
```diff
@@ -35,7 +35,18 @@ impl WeighMultiAssets for MultiAssetFilter {
...
+				All => weight.saturating_mul(MAX_ASSETS as u64),
+				AllOf { fun, .. } => match fun {
+					WildFungibility::Fungible => weight,
+					// Magic number 2 has to do with the fact that we could have up to 2 times
+					// MaxAssetsIntoHolding in the worst-case scenario.
+					WildFungibility::NonFungible =>
+						weight.saturating_mul((MaxAssetsIntoHolding::get() * 2) as u64),
+				},
+				AllCounted(count) => weight.saturating_mul(*count as u64),
+				AllOfCounted { count, .. } => weight.saturating_mul(*count as u64),
```
Comment (KiChjang) [inline parachains/runtimes/assets/westmint/src/weights/xcm/mod.rs:48]:
```suggestion
				AllCounted(count) => weight.saturating_mul(MAX_ASSETS.min(*count as u64)),
				AllOfCounted { count, .. } => weight.saturating_mul(MAX_ASSETS.min(*count as u64)),
```

File: parachains/runtimes/bridge-hubs/bridge-hub-kusama/src/weights/xcm/mod.rs [hunk 4]
```diff
@@ -35,7 +35,18 @@ impl WeighMultiAssets for MultiAssetFilter {
...
+				All => weight.saturating_mul(MAX_ASSETS as u64),
+				AllOf { fun, .. } => match fun {
+					WildFungibility::Fungible => weight,
+					// Magic number 2 has to do with the fact that we could have up to 2 times
+					// MaxAssetsIntoHolding in the worst-case scenario.
+					WildFungibility::NonFungible =>
+						weight.saturating_mul((MaxAssetsIntoHolding::get() * 2) as u64),
+				},
+				AllCounted(count) => weight.saturating_mul(*count as u64),
+				AllOfCounted { count, .. } => weight.saturating_mul(*count as u64),
```
Comment (KiChjang) [inline parachains/runtimes/bridge-hubs/bridge-hub-kusama/src/weights/xcm/mod.rs:48]:
```suggestion
				AllCounted(count) => weight.saturating_mul(MAX_ASSETS.min(*count as u64)),
				AllOfCounted { count, .. } => weight.saturating_mul(MAX_ASSETS.min(*count as u64)),
```

File: parachains/runtimes/bridge-hubs/bridge-hub-polkadot/src/weights/xcm/mod.rs [hunk 5]
```diff
@@ -35,7 +35,18 @@ impl WeighMultiAssets for MultiAssetFilter {
...
+				All => weight.saturating_mul(MAX_ASSETS as u64),
+				AllOf { fun, .. } => match fun {
+					WildFungibility::Fungible => weight,
+					// Magic number 2 has to do with the fact that we could have up to 2 times
+					// MaxAssetsIntoHolding in the worst-case scenario.
+					WildFungibility::NonFungible =>
+						weight.saturating_mul((MaxAssetsIntoHolding::get() * 2) as u64),
+				},
+				AllCounted(count) => weight.saturating_mul(*count as u64),
+				AllOfCounted { count, .. } => weight.saturating_mul(*count as u64),
```
Comment (KiChjang) [inline parachains/runtimes/bridge-hubs/bridge-hub-polkadot/src/weights/xcm/mod.rs:48]:
```suggestion
				AllCounted(count) => weight.saturating_mul(MAX_ASSETS.min(*count as u64)),
				AllOfCounted { count, .. } => weight.saturating_mul(MAX_ASSETS.min(*count as u64)),
```

File: parachains/runtimes/bridge-hubs/bridge-hub-rococo/src/weights/xcm/mod.rs [hunk 6]
```diff
@@ -35,7 +35,18 @@ impl WeighMultiAssets for MultiAssetFilter {
...
+				All => weight.saturating_mul(MAX_ASSETS as u64),
+				AllOf { fun, .. } => match fun {
+					WildFungibility::Fungible => weight,
+					// Magic number 2 has to do with the fact that we could have up to 2 times
+					// MaxAssetsIntoHolding in the worst-case scenario.
+					WildFungibility::NonFungible =>
+						weight.saturating_mul((MaxAssetsIntoHolding::get() * 2) as u64),
+				},
+				AllCounted(count) => weight.saturating_mul(*count as u64),
+				AllOfCounted { count, .. } => weight.saturating_mul(*count as u64),
```
Comment (KiChjang) [inline parachains/runtimes/bridge-hubs/bridge-hub-rococo/src/weights/xcm/mod.rs:48]:
```suggestion
				AllCounted(count) => weight.saturating_mul(MAX_ASSETS.min(*count as u64)),
				AllOfCounted { count, .. } => weight.saturating_mul(MAX_ASSETS.min(*count as u64)),
```

File: parachains/runtimes/bridge-hubs/bridge-hub-rococo/src/weights/xcm/mod.rs [hunk 7]
```diff
@@ -28,14 +28,25 @@ trait WeighMultiAssets {
...
+const MAX_ASSETS: u64 = 100;
 
 impl WeighMultiAssets for MultiAssetFilter {
 	fn weigh_multi_assets(&self, weight: Weight) -> Weight {
 		match self {
 			Self::Definite(assets) =>
 				weight.saturating_mul(assets.inner().into_iter().count() as u64),
-			Self::Wild(_) => weight.saturating_mul(MAX_ASSETS as u64),
+			Self::Wild(asset) => match asset {
+				All => weight.saturating_mul(MAX_ASSETS as u64),
```
Comment (KiChjang) [inline parachains/runtimes/bridge-hubs/bridge-hub-rococo/src/weights/xcm/mod.rs:39]:
```suggestion
				All => weight.saturating_mul(MAX_ASSETS),
```

**Review by KiChjang — APPROVED** [review]

File: parachains/runtimes/bridge-hubs/bridge-hub-rococo/src/weights/xcm/mod.rs (hunk 7 above)
Comment (KiChjang) [reply parachains/runtimes/bridge-hubs/bridge-hub-rococo/src/weights/xcm/mod.rs:39]: Same for the rest.

**Comment by EgorPopelyaev** [comment]
bot merge

**Pull request merged**